### PR TITLE
feat: add xorm closeState and sqlite's randomName

### PR DESCRIPTION
**Pull request by Malyue**
#### What this PR does / why we need it:
To add the DataSourceName and Close methods for xorm, and support SQLite3's random filename,

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reviewers:
/assign @sfwn 

#### CheckList
- [x] This pr has already passed build checks in erda
- [x] This pr has already passed by unit-test

### CODE REVIEW

File: providers/mysqlxorm/sqlite3/interface.go [hunk 1]
```diff
@@ -74,7 +92,29 @@ func NewSqlite3(dbSourceName string, opts ...OptionFunc) (*Sqlite3, error) {
...
+		}
+
+		err = os.Remove(s.DataSourceName())
+		return err
+	}
+	return nil
+}
+
+func randomName(path string) string {
+	dir, file := filepath.Split(path)
```
Comment (sfwn) [inline providers/mysqlxorm/sqlite3/interface.go:116]: use uuid directly or obey the rule of `os.CreateTemp`.

File: providers/mysqlxorm/xorm.go [hunk 2]
```diff
@@ -34,6 +35,9 @@ import (
 type Interface interface {
 	DB() *xorm.Engine
 	NewSession(ops ...SessionOption) *Session
+	Close() error
+	GetCloseState() bool
```
Comment (sfwn) [inline providers/mysqlxorm/xorm.go:39]: it's unnecessary to expose `GetCloseState` method. The `Close` method can be invoked multiple times idempotently.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: providers/mysqlxorm/sqlite3/interface.go (hunk 1 above)
Comment (Malyue) [reply providers/mysqlxorm/sqlite3/interface.go:108]: Done.

File: providers/mysqlxorm/xorm.go (hunk 2 above)
Comment (Malyue) [reply providers/mysqlxorm/xorm.go]: Done.

**Review by sfwn — APPROVED** [review]

**Pull request merged**